docs/next/guides/building-and-testing/plugins.md

```diff
@@ -17,228 +15,242 @@ tags:
+
   /**
    * The afterChange hook callback.
    *
-   * @param {Array} changes Array of changes.
-   * @param {String} source Describes the source of the change.
+   * @param {CellChange[]} changes An array of changes.
+   * @param {string} source Describes the source of the change.
    */
   onAfterChange(changes, source) {
     // afterChange callback goes here.
+    console.log(
+      `${CustomPlugin.PLUGIN_KEY}.onAfterChange - ${this.configuration.msg}`,
+      changes,
+      source
+    );
   }
 
   /**
-   * The destroy method should de-assign all of your properties.
+   * The `destroy` method is the best place to clean up any instances,
+   * objects or index mappers created during the plugin's lifecycle.
    */
   destroy() {
-    // The super method takes care of de-assigning the event callbacks, plugin hooks and clearing all the plugin properties.
+    // The super method takes care of de-assigning the event callbacks,
+    // plugin hooks and clearing all the plugin properties.
     super.destroy();
   }
 }
-
-export {InternalPluginSkeleton};
-
-// You need to register your plugin in order to use it within Handsontable.
-registerPlugin('internalPluginSkeleton', InternalPluginSkeleton);
-
 ```
 
-### External plugin
-
-Similarly to the above, this template is [hosted on GitHub](https://github.com/handsontable/handsontable-skeleton/tree/master/plugins/external).
-
-```js
-/**
- * @plugin External plugin skeleton.
- * Note: keep in mind, that Handsontable instance creates one instance of the plugin class.
- *
- * @param hotInstance
- * @constructor
- */
-function ExternalPluginSkeleton(hotInstance) {
-
-  // Call the BasePlugin constructor.
-  Handsontable.plugins.BasePlugin.call(this, hotInstance);
+### 3. Register CustomPlugin
+Last but not least, you need to register the plugin. This is the only way to use plugins in Handsontable.
+There are two ways to do that.
 
-  this._superClass = Handsontable.plugins.BasePlugin;
+- You can define a static getter named `PLUGIN_KEY` that the `registerPlugin` utility uses as the plugin's alias. Usage example you can find in the above code snippet.
+  ```js
+  // You need to register your plugin in order to use it within Handsontable.
+  registerPlugin(CustomPlugin);
+  ```
+- You can also use the custom alias. Put a string in the first argument. The registerer will use that string as an alias instead of the `PLUGIN_KEY` getter from `CustomPlugin`.
+  ```js
+  registerPlugin('CustomAlias', CustomPlugin);
+  ```
 
-  // Initialize all your public properties in the class' constructor.
-  /**
-   * yourProperty description.
-   *
-   * @type {String}
-   */
-  this.yourProperty = '';
-  /**
-   * anotherProperty description.
-   * @type {Array}
-   */
-  this.anotherProperty = [];
-}
+### 4. Use plugin in Handsontable
+Accordingly, to the supported options for the plugin, you might control it, passing in its options the boolean or object.
+The following snippet presents all possible approaches:
 
```

Comment:
\### 4. Use your plugin in Handsontable
To control the plugin's options, pass a boolean or an object at the plugin's initialization: